presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -391,15 +395,21 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     @Override
     public void createSchema(ConnectorSession session, String schemaName, Map<String, Object> properties)
     {
-        Optional<String> location = HiveSchemaProperties.getLocation(properties).map(locationUri -> {
-            try {
-                hdfsEnvironment.getFileSystem(session.getUser(), new Path(locationUri));
-            }
-            catch (IOException e) {
-                throw new PrestoException(INVALID_SCHEMA_PROPERTY, "Invalid location URI: " + locationUri, e);
-            }
-            return locationUri;
-        });
+        Optional<String> customLocation = HiveSchemaProperties.getLocation(properties);
+        Optional<String> defaultLocation = Optional.of(this.metastoreWarehouseDir + "/" + schemaName + ".db");
+
+        Optional<String> location = Stream.of(customLocation, defaultLocation)
+                .filter(Optional::isPresent)
+                .findFirst()
+                .flatMap(locationUri -> {
+                    try {
+                        hdfsEnvironment.getFileSystem(session.getUser(), new Path(locationUri.get()));
```

Comment:
This change seems unrelated to storage based security. Can you pull it out into a separate commit or PR?

Why is this needed? The metastore should be creating the location. We intentionally avoided having a default location config in Presto since that should be up to the metastore.